web/src/composebox_typeahead.ts

```diff
@@ -1136,8 +1138,29 @@ export function content_typeahead_selected(
             // Stream + topic mention typeahead; close the stream+topic mention syntax
             // with the topic and the final **.  Note that token.length can be 0
             // if we are completing from `**streamname>`.
+            function will_produce_broken_link(topic: string): boolean {
+                return /(\*+)|(.*`.*`.*)/.test(topic);
+            }
+            function get_stream_name(syntax: string): string {
+                const start = syntax.lastIndexOf("#**");
+                const end = syntax.lastIndexOf(">");
+                return syntax.slice(start + 3, end);
+            }
+            function url_syntax(stream_name: string, topic_name: string): string {
+                const stream_id = helpers.get_stream_by_name(stream_name)?.stream_id ?? -1;
+                return `[#${stream_name}>${topic_name}](${helpers.stream_topic_hash(stream_id, topic_name)})`;
+            }
+            // If the topic name is such that it will generate an invalid #**stream>topic** syntax,
+            // revert to generating the normal markdown syntax for a link
             const start = beginning.length - token.length;
             beginning = beginning.slice(0, start) + item.topic + "** ";
+            const syntax_start_index = beginning.lastIndexOf("#**");
+            if (will_produce_broken_link(item.topic)) {
+                beginning =
+                    beginning.slice(0, syntax_start_index) +
+                    url_syntax(get_stream_name(beginning), item.topic) +
+                    " ";
+            }
```

Comment:
Can you extract this logic to a function and/or add unit tests for it? It's complicated enough that it's very much worth having a test for.